src/flashbar/collapsible-flashbar.tsx

```diff
@@ -139,16 +146,29 @@ export default function CollapsibleFlashbar({ items, ...restProps }: FlashbarPro
 
     if (initialAnimationState) {
       updateBottomSpacing();
+
+      // When collapsing with the keyboard, scroll up if necessary to avoid losing track of the focused button
+      const shouldScrollUp =
+        notificationBarRef.current &&
+        !isFlashbarStackExpanded &&
+        isKeyboardInteraction(isFocusVisible) &&
+        isElementTopBeyondViewport(notificationBarRef.current);
+
       animate({
         elements: getElementsToAnimate(),
         oldState: initialAnimationState,
         newElementInitialState: ({ top }) => ({ scale: 0.9, y: -0.2 * top }),
         onTransitionsEnd: () => setTransitioning(false),
       });
+
+      if (shouldScrollUp) {
+        window.scrollTo({ top: 0 });
```

Comment:
If your task is to scroll button into viewport, use `button.scrollIntoViewport` ([our wrapper with better defaults](https://github.com/cloudscape-design/components/blob/657d369dc08468d2066cd64b12ad93804d6cc911/src/internal/utils/scrollable-containers.ts#L76-L81)) and not manual scroll on the window. There could be multiple scrollable containers in a way, this code is bug-prone.


-- 

Also, why is there a conditionality on keyboard/mouse interactions? They both should receive the same scroll treatment

Reply:
Consulting with a11y and did both changes:
- Use `scrollIntoViewport` with existing function
- Apply the same behavior with mouse as with keyboard

Manually tested again and updated unit tests accordingly.